src/SourceTerms/Leakage/CPU_Src_Leakage_ComputeLeak.cpp

```diff
@@ -0,0 +1,1154 @@
+                              const bool NuHeat, const real NuHeat_Fac, const real UNIT_D, const EoS_t *EoS );
+
+GPU_DEVICE static
+real Compute_FermiIntegral( const int order, const real eta );
+
+
+
+
+#ifndef __CUDACC__
+//-------------------------------------------------------------------------------------------------------
+// Function    :  Src_Leakage_ComputeTau
+// Description :  Compute the neutrino optical depth using the Ruffert and Rosswog approaches,
+//                and the neutrino flux, mean/rms neutrino energy for neutrino heating
+//
+// Note        :  1. Invoked by Src_WorkBeforeMajorFunc_Leakage()
+//                2. The stored Heat_Flux is scaled by Const_hc_MeVcm_CUBE to avoid overflow
+//                3. Add "#ifndef __CUDACC__" since this routine is only useful on CPU
+//                4. Ref: M. Rufferrt, H.-Th. Janka, G. Schaefer, 1996, A&A, 311, 532 (arXiv: 9509006)
+//                        S. Rosswog & M. Liebendoerfer, 2003, MNRAS, 342, 673 (arXiv: 0302301)
+//                        E. O'Connor & C. D. Ott, 2010, Class. Quantum Grav., 27, 114103 (arXiv: 0912.2393)
+//
+// Parameter   :  Ray       : Profile_t object array stores the profiles of all rays
+//                            --> Density     : in code unit
+//                            --> Temperature : in Kelvin
+//                            --> Ye          : dimensionless
+//                Edge      : Edge of bins in the radial direction in code unit
+//                NRadius   : Number of bins in the radial    direction
+//                NTheta    : Number of bins in the polar     direction
+//                NPhi      : Number of bins in the azimuthal direction
+//                tau_Ruff  : Array to store the opacity calculated via the Ruffert scheme
+//                chi_Ross  : Array to store the chi     calculated via the Rosswog scheme
+//                Heat_Flux : Array to store the      neutrino flux used for heating
+//                HeatE_Rms : Array to store the rms  neutrino energy at neutrino sphere
+//                HeatE_Ave : Array to store the mean neutrino energy at neutrino sphere
+//
+// Return      :  tau_Ruff, chi_Ross, Heat_Flux, HeatE_Rms, HeatE_Ave
+//-------------------------------------------------------------------------------------------------------
+void Src_Leakage_ComputeTau( Profile_t *Ray[], double *Edge,
+                             const int NRadius, const int NTheta, const int NPhi,
+                             real tau_Ruff [][NTheta*NPhi][NType_Neutrino],
+                             real chi_Ross [][NTheta*NPhi][NType_Neutrino],
+                             real Heat_Flux[][NTheta*NPhi][NType_Neutrino],
+                             real HeatE_Rms[][NType_Neutrino],
+                             real HeatE_Ave[][NType_Neutrino] )
+{
+
+   #ifdef FLOAT8
+   const double Tolerance    = 1.0e-10;
+   #else
+   const double Tolerance    = 1.0e-6;
+   #endif
```

Comment:
I think the tolerance is already defined in `NuclearEoS.h` which is for Newton-Raphson or bisection though. Please check the indentations also.

Reply:
Rename to `Tolerance_Leak` and set to `1e-10`.